contracts-proc/src/interface.rs

```diff
@@ -0,0 +1,118 @@
+use std::mem;
+
+use convert_case::{Case, Casing};
+use proc_macro::TokenStream;
+use proc_macro2::Ident;
+use quote::quote;
+use syn::{
+    parse::{Parse, ParseStream},
+    parse_macro_input, FnArg, ItemTrait, LitStr, Result, Token, TraitItem,
+};
+
+/// Computes interface id as an associated constant for the trait.
+pub(crate) fn interface(_attr: TokenStream, input: TokenStream) -> TokenStream {
```

Comment:
Maybe `erc165_interface` is a better name. Why? There might be other type of interface like macros we want to add in the future or the sdk itself might add one with this name

Reply:
Totally agree with @ggonzalez94 in terms of better name for the macro attribute. For me it is not enough self-explanatory. I would name `#[erc165]`

Reply:
decided to go with `#[interface_id]` macro name